docs/src/explanations/naming.md

```diff
@@ -0,0 +1,214 @@
+---
+layout: docs
+title:  "Naming"
+section: "chisel3"
+---
+
+Historically, Chisel has had trouble reliably capturing the names of signals. The reasons for this are due to (1)
+primarily relying on reflection to find names, (2) using `@chiselName` macro which had unreliable behavior, and (3)
+when all else fails using the `.suggestName("foo")` API.
```

Comment:
this isn't exactly a "reason" for why Chisel couldn't reliably capture the names of signals